test/unit/logic/transaction_pool.js

```diff
@@ -110,27 +170,1073 @@ describe('transactionPool', () => {
+					expect(err).to.be.null;
+					expect(res).to.deep.eql(transaction);
+					done();
+				})
+			);
+		});
+	});
+
+	describe('processBundled', () => {
+		let processVerifyTransaction;
+		beforeEach(done => {
+			resetStates();
+			transactionPool.addBundledTransaction({ id: '123', bundled: true });
+			processVerifyTransaction = TransactionPool.__get__(
+				'__private.processVerifyTransaction'
+			);
+			done();
+		});
+
+		afterEach(done => {
+			TransactionPool.__set__(
+				'__private.processVerifyTransaction',
+				processVerifyTransaction
+			);
+			done();
+		});
+
+		it('should fail to process the bundle transaction', done => {
+			const processVerifyTransactionStub = sinon
+				.stub()
+				.callsArgWith(2, 'Failed to process bundle transaction');
+			TransactionPool.__set__(
+				'__private.processVerifyTransaction',
+				processVerifyTransactionStub
+			);
+			transactionPool.processBundled(() => {
+				expect(logger.debug.args[0][0]).to.deep.eql(
+					'Failed to process / verify bundled transaction: 123'
+				);
+				expect(logger.debug.args[0][1]).to.deep.eql(
+					'Failed to process bundle transaction'
+				);
+				done();
+			});
+		});
+
+		it('should fail to queue bundled transaction', done => {
+			const processVerifyTransactionStub = sinon.stub().callsArgWith(2, null);
+			TransactionPool.__set__(
+				'__private.processVerifyTransaction',
+				processVerifyTransactionStub
+			);
+			transactionPool.queueTransaction.callsArgWith(
+				1,
+				'Failed to queue bundled transaction'
+			);
+			transactionPool.processBundled(() => {
+				expect(logger.debug.args[0][0]).to.deep.eql(
+					'Failed to queue bundled transaction: 123'
+				);
+				expect(logger.debug.args[0][1]).to.deep.eql(
+					'Failed to queue bundled transaction'
+				);
+				done();
+			});
+		});
+	});
+
+	describe('undoUnconfirmedList', () => {
+		let undoUnconfirmedList;
+		let lastError;
+		let lastIds;
+
+		before(done => {
+			undoUnconfirmedList = transactionPool.undoUnconfirmedList;
+			done();
+		});
+
+		describe('when unconfirmed lists', () => {
+			describe('is empty', () => {
+				const transactions = [];
+
+				before(done => {
+					transactionPool.getUnconfirmedTransactionList = function() {
+						return transactions;
+					};
+
+					undoUnconfirmedList((err, ids) => {
+						lastError = err;
+						lastIds = ids;
+						done();
+					});
+				});
+
+				it('should not return an error', () => {
+					return expect(lastError).to.not.exist;
+				});
+
+				it('should not log an error', () => {
+					return expect(logger.error.called).to.be.false;
+				});
+
+				it('should return empty ids array', () => {
+					expect(lastIds).to.be.an('array');
+					return expect(lastIds.length).to.equal(0);
+				});
+
+				describe('modules.transactions.undoUnconfirmed', () => {
+					it('should not be called', () => {
+						return expect(dummyUndoUnconfirmed.called).to.be.false;
+					});
+				});
+
+				after(resetStates);
```

Comment:
No need to resetStates when beforeEach creates the object and afterEach restore sinonSandbox